browse/services/documents/db_implementation/db_abs.py

```diff
@@ -0,0 +1,140 @@
+            secondary_categories=latest_version.secondary_categories,
+            primary_archive=latest_version.primary_archive,
+            primary_group=latest_version.primary_group,
+            is_definitive=True,
+            is_latest=False)
+        return combined_version
+
+
+    def _abs_for_version(self, identifier: Identifier,
+                         version: Optional[int] = None) -> DocMetadata:
+        """Get a specific version of a paper's abstract metadata.
+
+        if version is None then get the latest version."""
+        if version:
+            res = (Metadata.query
+                   .filter( Metadata.paper_id == identifier.id)
+                   .filter( Metadata.version == identifier.version )).first()                   
+        else:
+            res = (Metadata.query
+                   .filter(Metadata.paper_id == identifier.id)
+                   .filter(Metadata.is_current == 1)).first()
+        if not res:
+            raise AbsNotFoundException(identifier.id)
+        return to_docmeta(res)
+    
+    def get_dissemination_formats(self,
+                                  docmeta: DocMetadata,
+                                  format_pref: Optional[str] = None,
+                                  add_sciencewise: bool = False) -> List[str]:
+        """Get a list of formats that can be disseminated for this DocMetadata.
+
+        THIS ONLY CHECK THE source type on the doc metadata.
+
+        Format names are strings. These include 'src', 'pdf', 'ps', 'html',
+        'pdfonly', 'other', 'dvi', 'ps(400)', 'ps(600)', 'nops'.
+
+        Parameters
+        ----------
+        docmeta : :class:`DocMetadata`
+        format_pref : str
+            The format preference string.
+        add_sciencewise : bool
+            Specify whether to include 'sciencewise_pdf' format in list.
+
+        Returns
+        -------
+        List[str]
+            A list of format strings.
+        """
+        # version = docmeta.version
+        # format_code = docmeta.version_history[version - 1].source_type.code
+
+        # # TODO cache flag?
+        # cache_flag = False
+        # # cached_ps_file_path = cache.get_cache_file_path(docmeta, 'ps')
+        # # if cached_ps_file_path \
+        # #    and os.path.getsize(cached_ps_file_path) == 0 \
+        # #    and source_file_path \
+        # #    and os.path.getmtime(source_file_path) \
+        # #    < os.path.getmtime(cached_ps_file_path):
+        # #     cache_flag = True
+
+        # return formats_from_source_type(format_code,
+        #                                 format_pref,
+        #                                 cache_flag,
+        #                                 add_sciencewise)
+        return []
+
+    def get_ancillary_files(self, docmeta: DocMetadata) \
+            -> List[Dict]:
+        """Get list of ancillary file names and sizes."""
+        # TODO implement get_ancillary_files
```

Comment:
reminder for later, per TODO

Reply:
I've created an issue for this https://arxiv-org.atlassian.net/browse/ARXIVNG-4283